src/otx/v2/adapters/torch/dataset.py

```diff
@@ -31,20 +31,20 @@ def _initialize(self) -> None:
     def _build_dataset(
         self,
         subset: str,
-        pipeline: list | dict | None = None,
-        config: str | dict | None = None,
+        pipeline: list | None = None,
+        config: dict | None = None,
     ) -> TorchDataset | None:
         """Builds a TorchDataset object for the given subset using the specified pipeline and configuration.
 
         Args:
             subset (str): The subset to build the dataset for.
-            pipeline (Optional[Union[list, dict]]): The pipeline to use for the dataset.
+            pipeline (list | None, optional): The pipeline to use for the dataset.
```

Comment:
Isn't this a duplicate? I'm not sure about the convention, but it would be better, IMHO, to be either of

- pipeline (list, optional)
- pipeline (list | None)

Please ignore if it's well accepted convention. :)

Reply:
I'm not sure of the exact convention, but I understand the meaning of optional to be whether it has a default value or not.

Reply:
I'll have to look into the convention a bit more and try to make it consistent across the entire code (in the near future..?)